static/app/app/controller.js

```diff
@@ -59,6 +60,34 @@ var Controller = Marionette.Object.extend({
             spots.fetch().done(callback);
         }
 
+    },
+    join: function(moveId) {
+        var moves = channel.request('entities:moves');
+        
+        $.when(moves.fetch()).done(function(){
+            var currentMove = channel.request("entities:move");
+            var moveToJoin = channel.request('entities:moves')
+                                    .find({id: parseInt(moveId)});
+            
+            currentMove.set('time', moveToJoin.get('time').format());
+            currentMove.set('spot', moveToJoin.get('spot'));
+            
+            if (currentMove.get('user')) {
+                currentMove.save({}, {
+                    dataType: 'text',
+                    success: function(model, resp) {
+                        channel.trigger('list');
+                        console.log(model);
+                    },
+                    error: function(model, resp) {
+                        console.log(resp);
+                    }
+                })
+            } else {
+                var mainRegion = channel.request('get:region', 'main');
+                mainRegion.show(new JoinView);
```

Comment:
While it is definitely good thinking to try to streamline the process and only makes users to add in their name if they go through the join route, I think it would be better to just trigger the `edit` route here. I don't think the minor annoyance in UX of having to hit save twice really merits creating a separate view with mostly duplicate logic.